website/client-old/js/services/questServices.js

```diff
@@ -104,6 +104,9 @@ angular.module('habitrpg')
             Groups.data.party = party;
             $state.go('options.social.party');
             resolve();
+            if ($state.current.name === "options.social.party") {
+              $state.reload();
```

Comment:
I'm unsure if you can do this.

The chain of command from the party screen starts in the [PartyCtrl](https://github.com/HabitRPG/habitica/blob/develop/website/client-old/js/controllers/partyCtrl.js#L157): If we have quests we load the InventoryCtrl, if not we switch to the Inventory state (see other chain of command). From the InventoryCtrl, the [ownedQuests jade file](https://github.com/HabitRPG/habitica/blob/a4fd687510160fc54e5730266f4f921f63f49c53/website/views/shared/modals/quests.jade#L126) is constructed and placed into the quest popup. Then we deal with the logic that I changed.

In the chain of command from the Inventory side, it's similar but we don't have the Party Ctrl.

If we move the "$state.reload" into the party controller, it doesn't reload after starting a quest( I tried this out). Probably due to the fact that the chain of command doesn't finish before it reloads, due to the popup quest selection window.

Moving it into the InventoryCtrl doesn't quite make sense/work, since it's still used by both chains of command so it would still require a conditional.

I may be wrong about some of this but I do think the logic should be there. I am also moderaely unfamiliar with the codebase as a whole though.